test/functional/page_objects/home_page.ts

```diff
@@ -91,40 +91,46 @@ export class HomePageObject extends FtrService {
 
   async addSampleDataSet(id: string) {
     await this.openSampleDataAccordion();
-    const isInstalled = await this.isSampleDataSetInstalled(id);
-    if (!isInstalled) {
+    await this.retry.waitFor('sample data to be installed', async () => {
+      // count for the edge case where some how installation completes just before the retry occurs
+      if (await this.isSampleDataSetInstalled(id)) {
+        return true;
+      }
+
       this.log.debug(`Attempting to add sample data: ${id}`);
-      await this.retry.waitFor('sample data to be installed', async () => {
-        // Echoing the adjustments made to 'removeSampleDataSet', as we are seeing flaky test cases here as well
-        // https://github.com/elastic/kibana/issues/52714
-        await this.testSubjects.waitForEnabled(`addSampleDataSet${id}`);
-        await this.common.sleep(1010);
-        await this.testSubjects.click(`addSampleDataSet${id}`);
-        await this.common.sleep(1010);
-        await this._waitForSampleDataLoadingAction(id);
-        return await this.isSampleDataSetInstalled(id);
-      });
-    }
+
+      // Echoing the adjustments made to 'removeSampleDataSet', as we are seeing flaky test cases here as well
+      // https://github.com/elastic/kibana/issues/52714
+      await this.testSubjects.waitForEnabled(`addSampleDataSet${id}`);
+      await this.common.sleep(1010);
+      await this.testSubjects.click(`addSampleDataSet${id}`);
+      await this.common.sleep(1010);
+      await this._waitForSampleDataLoadingAction(id);
+      return await this.isSampleDataSetInstalled(id);
+    });
   }
```

Comment:
Changed this logic, because it actually triggers the flakiness of the test, looking at the [logs](https://buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/b1106efc-ba5e-4e90-a857-11c4a899c05d/0192ba20-25b2-472d-8814-95b6ded886ca/0192ba55-f39f-4887-b712-0c5eb0e9a709/target/test_failures/0192ba55-f39f-4887-b712-0c5eb0e9a709_2d57a563cfc9368ff2eb5653c9051b78.log?response-content-type=text%2Fplain&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LQCVXXWXF%2F20241105%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241105T144500Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJHMEUCIFR4KyxyzwG53AFQaQder0ye1bMFS72kRUcKuTaN5cNnAiEAh60i2SQ7T3jSB8cgDOvWWCgJpDOVzdszfVPEVbiXyWYq8QMIFRAAGgwwMzIzNzk3MDUzMDMiDPudDPPdsZtBCh8ERCrOA0akLcgw8sn7pJMGbfiUf6jBWAat1WmKVAdULcCsZBS%2BukAMH2gxJbS9M%2F4pHPZFhTBVfBuagm3GFEQy5gCyXrY4ljUg2XA9xP3fm3UN7Oq3%2FSu33m59zkPuEDSpgsRDXi4JHhPw0r61ZevsEwivpA1aRTPtKIGPlE9tnTpzFDOwLsT86FG6Ew93EQ7cVThhbsOl3c12DC3qXgyLYeKgiz6ELSfxiclvLh7zpZptepfl6GgI451X0IXmXktZmkArBMZCfAvxsABvzBRY9qrFp26o4lHt2rYklcx8sSnaTDauI1B%2FBfxncqD5eaJhaYcJh84eearLxCvCvFIcOxnEwzFt55kIjR6FEKcGE7pxnCXA%2F3Ex3UHIMioqxa6%2BkU48jg%2Fn03U1SxQ22eKodR5dZHaYkb37E50lHh5%2BintQT3oinTC%2BBrY7tNs4lnGKPRWoKXev4jNA3%2FPrS5y%2FEB06KGHMy0TJ6HZ%2BliNnfdP8Ey0DCRn6UqDmt1rbYR7QrZWqLCV2pYL2QmxzUAB2uyrebEuLm0SPQ4K9hSlJq8h4GZxuOHebV9zYm2NUWZF9WgSXWj%2FDbx104QmpTGVWYQeUmiGOSiD%2FeuoEOlPYfjKYDDDKjKi5BjqlAXMT7yWe397AA4Hng5%2Fjl5jOJGNSy%2BaVgQvgL6DLPf7Cq8xnj60uRMP%2BSUkjfRzRPUL%2Fsutcpeb6HuHO5xADQIAodW4Ziw%2BLjKUVMDCKzPQCPZ0tUySsS2jeX7ujNB%2Babiz6ogECI4i9RDuIGMvErkn3A1zYvNjJ1E1Q8U0AkQnnmRRCDZQZ6%2BOzJ4HDqjdMGHyoU%2BgRG2bUwICg4cSnqRiNuAcv2Q%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=19dacc0ee0c7557bbb50692e9ae6c061053e0fc74a79368fb51291b145e363cc) of a previous run, we see the sample data installation completes, however for some reason at the point whilst the installation status was checked a falsy value was returned which  in turn recursively initiated to complete an install that's no longer possible, hence the reason for adopting an early return once the desired state exists.